tpp-app/src/home/service/utils/helpers.js

```diff
@@ -0,0 +1,18 @@
+// Backend helper functions used across app
+
+/**
+ * Initializes an empty year array with 12 nested arrays, representing a month.
+ * Within each month array is X null values corresponding to X days of that month in that year.
+ * @param yearNumber number
+ */
+const initializeEmptyYear = (yearNumber) => {
+  let year = new Array(12);
+
+  for (let i = 0; i < 12; ++i) {
+    let daysInMonth = new Date(yearNumber, i+1, 0).getDate();
+    let month = new Array(daysInMonth).fill(null); // fill with daysInMonth null values
```

Comment:
I agree! Don't think `.fill(null)` is necessary.

Reply:
yeah it's just that in all our documentation it's `null` LOL

Reply:
@yichenesia @faye-tan maybe y'all could provide some input?

Reply:
Oh yeah if truth value checking is identical then yeah that totally makes sense

Reply:
ok either null or undefined but we should choose one or the other otherwise we would have to watch out for weak equality and strong equality: 
null === undefined // false
null == undefined // true

Reply:
I'll keep it at `null` then since that's what's in the documentation.